cellular_automata_sgs.F90

```diff
@@ -42,93 +46,73 @@ subroutine cellular_automata(kstep,Statein,Coupling,Diag,nblks,nlev, &
 type(random_stat) :: rstate
-integer :: nlon, nlat, isize,jsize,nf,k350,k850
+integer :: nlon, nlat, isize,jsize,nf,nn
 integer :: inci, incj, nxc, nyc, nxch, nych
 integer :: halo, k_in, i, j, k, iec, jec, isc, jsc
 integer :: seed, ierr7,blk, ix, iix, count4,ih,jh
-integer :: blocksz,levs,ra,rb,rc
+integer :: blocksz,levs
 integer(8) :: count, count_rate, count_max, count_trunc
 integer(8) :: iscale = 10000000000
-integer, allocatable :: iini(:,:,:),ilives(:,:),ca_plumes(:,:)
+integer, allocatable :: iini(:,:,:),ilives(:,:,:),iini_g(:,:,:),ilives_g(:,:),ca_plumes(:,:)
 real(kind=kind_phys), allocatable :: field_out(:,:,:), field_in(:,:),field_smooth(:,:),Detfield(:,:,:)
-real(kind=kind_phys), allocatable :: omega(:,:,:),pressure(:,:,:),cloud(:,:),humidity(:,:)
-real(kind=kind_phys), allocatable :: vertvelsum(:,:),vertvelmean(:,:),dp(:,:,:),surfp(:,:)
-real(kind=kind_phys), allocatable :: CA(:,:),CAstore(:,:),CAavg(:,:),condition(:,:),rho(:,:),cape(:,:)
-real(kind=kind_phys), allocatable :: CA_DEEP(:,:),CA_TURB(:,:),CA_RAD(:,:),CA_MICRO(:,:),CA_SHAL(:,:)
+real(kind=kind_phys), allocatable :: omega(:,:,:),pressure(:,:,:),cloud(:,:),humidity(:,:),uwind(:,:)
+real(kind=kind_phys), allocatable :: vertvelsum(:,:),vertvelmean(:,:),dp(:,:,:),surfp(:,:),shalp(:,:),gamt(:,:)
+real(kind=kind_phys), allocatable :: CA(:,:),condition(:,:),rho(:,:),conditiongrid(:,:)
+real(kind=kind_phys), allocatable :: CA_DEEP(:,:),CA_TURB(:,:),CA_SHAL(:,:)
 real(kind=kind_phys), allocatable :: noise1D(:),vertvelhigh(:,:),noise(:,:,:)
-real(kind=kind_phys) :: psum,csum,CAmean,sq_diff,CAstdv,count1,alpha
-real(kind=kind_phys) :: Detmax(nca),Detmean(nca),phi,stdev,delt
+real(kind=kind_phys) :: psum,csum,CAmean,sq_diff,CAstdv,count1,lambda
+real(kind=kind_phys) :: Detmax(nca),Detmin(nca),Detmean(nca),phi,stdev,delt,condmax
 logical,save         :: block_message=.true.
 logical              :: nca_plumes
 
 !nca         :: switch for number of cellular automata to be used.
 !ca_global   :: switch for global cellular automata
-!ca_sgs      :: switch for cellular automata conditioned on SGS perturbed vertvel.
+!ca_sgs      :: switch for cellular automata conditioned on SGS perturbed vertvel. 
 !nfracseed   :: switch for number of random cells initially seeded
 !nlives      :: switch for maximum number of lives a cell can have
 !nspinup     :: switch for number of itterations to spin up the ca
-!ncells      :: switch for higher resolution grid e.g ncells=4
-!               gives 4x4 times the FV3 model grid resolution.
+!ncells      :: switch for higher resolution grid e.g ncells=4 
+!               gives 4x4 times the FV3 model grid resolution.                
 !ca_smooth   :: switch to smooth the cellular automata
 !nthresh     :: threshold of perturbed vertical velocity used in case of sgs
 !nca_plumes   :: compute number of CA-cells ("plumes") within a NWP gridbox.
 
-k350 = 29
-k850 = 13
-alpha = 1.5
-ra = 201
-rb = 2147483648
-rc = 4294967296
-
 halo=1
 k_in=1
 
```

Comment:
I suggest a code block such as:
if (nlev .EQ. 64) then
   k350=29
   k850=13
else (nlev. EQ. 127) then
  k350=61
  k850=28
else ! make a guess
   k350=nint(nlev/2)
   k850=nint(nlev/5)
   print*,'this level selection is not supported, making an approximation for k350 and k850'
endif

Reply:
Thanks, good suggestion!

Reply:
I addressed this and results are reproduced.

Reply:
good,  once you commit and push the updates, I can do the merge.